### PR TITLE
fix: clean members data

### LINKED ISSUE #34: Do not include sensitive information in export

### PR DESCRIPTION
**Pull request by pyphilia**
This PR adds a layer of validation for members, to prevent inclusion of sensitive info:
- password
- extra.favoriteItems

fix #34

### CODE REVIEW

File: src/services/action/base-analytics.test.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      itemMemberships,
+      metadata,
+    });
+
+    for (const m of analytics.members) {
+      // lang exists
+      expect(m.extra.lang).toBeTruthy();
+
+      // no password
+      expect(m.password).toBeFalsy();
```
Comment (dialexo) [inline src/services/action/base-analytics.test.ts:61]: Would this make sense?
```suggestion
      expect(m.password).toBeUndefined();
```

File: src/services/action/base-analytics.test.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+      // no favorites
+      expect(m.extra.favoriteItems).toBeFalsy();
```
Comment (dialexo) [inline src/services/action/base-analytics.test.ts:64]: Would this make sense?
```suggestion
      expect(m.extra.favoriteItems).toBeUndefined();
```

File: src/services/action/base-analytics.ts [hunk 3]
```diff
@@ -25,6 +27,31 @@ export class BaseAnalytics implements Analytics {
...
+          name: { type: 'string' },
+          email: { type: 'string' },
+          extra: {
+            type: 'object',
+            additionalProperties: false,
+            properties: { lang: { type: 'string' } },
+          },
+        },
+      },
+    };
```
Comment (dialexo) [inline src/services/action/base-analytics.ts:51]: Should this be moved to `src/schemas` and typed accordingly? (you then get static checks of your schema for free, so e.g. VSCode can tell you whether the schema is valid while you're typing it)

**Review by mochiruntime — APPROVED** [review]
LGTM, but feel free to apply my suggestions :crab:

**Pull request merged**